freqtrade/tests/test_configuration.py

```diff
@@ -617,3 +617,86 @@ def test_validate_tsl(default_conf):
     default_conf['trailing_stop_positive_offset'] = 0.015
     Configuration(Namespace())
     configuration._validate_config_consistency(default_conf)
+
+
+# config['exchange'] subtree has required options in it
+# so it cannot be omitted in the config
+def test_load_config_default_exchange(all_conf) -> None:
+    del all_conf['exchange']
+
+    assert 'exchange' not in all_conf
+
+    with pytest.raises(ValidationError,
+                       match=r'\'exchange\' is a required property'):
+        configuration = Configuration(Namespace())
+        configuration._validate_config_schema(all_conf)
+###    assert 'exchange' in all_conf
+
+
+# config['exchange']['name'] option is required
+# so it cannot be omitted in the config
+def test_load_config_default_exchange_name(all_conf) -> None:
+    del all_conf['exchange']['name']
+
+    assert 'name' not in all_conf['exchange']
+
+    with pytest.raises(ValidationError,
+                       match=r'\'name\' is a required property'):
+        configuration = Configuration(Namespace())
+        configuration._validate_config_schema(all_conf)
+
+
+# config['exchange']['sandbox'] option has default value: False
+# so it can be omitted in the config and the default value
+# should be present in the config as the option value
```

Comment:
If we need to describe what the comment does - please use docstrings... With them it's clearer "where" they belong - and having comments before a function is very irritating if we start using decorators...

``` python
def test_load_config_default_exchange_sandbox(all_conf) -> None:
    """
    config['exchange']['sandbox'] option has default value: False
    so it can be omitted in the config and the default value
    should be present in the config as the option value
    """
```

Reply:
Sorry, wrote too much in C previousy, just forgotten of this pythonic custom...

Reply:
Done.